scripts/InstancingWithModsSystem.js

```diff
@@ -0,0 +1,129 @@
+class InstancingSystem extends System {
+
+    // todo - title it InstancingWithModsSystem
+
+    constructor() {
+        super();
+
+        console.log(this.componentName);
+
+        const this.instanceCount = 5;
+
+        // DataOrientedDesign arrays
+        let this.positions = null;
+        let this.scales = null;
+        let this.rotations = null;
+        let this.meshes = [];
+        let this.geometry = null;
+        console.log('constructed an instancing system');
+    }
+
+    update(deltaTime, frame) {
+        for(const entity of this.registry){
+            switch (entity.components.get('instancing')?.type) {
+                case 'animate':
+                    this.animate(entity)
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    attachedComponent(entity, data) {
+        console.log('in attachedComponent');
+        console.log('entity is:');
+        console.log(entity);
+        console.log('data is:');
+        console.log(data);
+
+        this.createInstances();
+        console.log(entity.object3D);
+        geometry = entity.object3D.geometry;
```

Comment:
you're _probably_ gonna need to use [`getObjectByProperty`](https://threejs.org/docs/#api/en/core/Object3D.getObjectByName) here. models don't always load geometries.

Reply:
√